core/src/main/java/ai/timefold/solver/core/impl/score/director/ValueRangeManager.java

```diff
@@ -509,22 +766,32 @@ private static <T> void loadEntityValueRange(int entityIndex, Map<Object, Intege
             var valueIndex = valueIndexMap.get(value);
             var item = reachableValueList.get(valueIndex);
             item.addEntity(entityIndex);
-            for (var j = i + 1; j < valueRange.getSize(); j++) {
-                var otherValue = valueRange.get(j);
-                if (otherValue == null) {
-                    continue;
-                }
-                var otherValueIndex = valueIndexMap.get(otherValue);
-                var otherValueItem = reachableValueList.get(otherValueIndex);
-                item.addValue(otherValueIndex);
-                otherValueItem.addValue(valueIndex);
-            }
+            // We unset the current value index to import only the values that are reachable
+            allValuesBitSet.clear(valueIndex);
+            item.addValues(allValuesBitSet);
+            // We set it again to restore the state for the next value
+            allValuesBitSet.set(valueIndex);
         }
     }
 
     public void reset(@Nullable Solution_ workingSolution) {
```

Comment:
I question the need for this.
This is a very expensive thing. All that array filling, all that iteration.
Only to do the same thing that the GC would do as well, for almost free.